### PR TITLE
AcctIdx: age adds time component

### PR DESCRIPTION
**Pull request by jeffwashington**
#### Summary of Changes

Fixes #

### CODE REVIEW

File: runtime/src/bucket_map_holder.rs [hunk 1]
```diff
@@ -3,18 +3,24 @@ use crate::bucket_map_holder_stats::BucketMapHolderStats;
 use crate::in_mem_accounts_index::SlotT;
 use crate::waitable_condvar::WaitableCondvar;
 use solana_bucket_map::bucket_map::{BucketMap, BucketMapConfig};
+use solana_sdk::timing::AtomicInterval;
 use std::fmt::Debug;
 use std::sync::atomic::{AtomicBool, AtomicU8, AtomicUsize, Ordering};
 use std::sync::Mutex;
 pub type Age = u8;
 
+const AGE_MS: u64 = 400; // 400 ms per age increment, which matches slot time
```
Comment (brooksprumo) [inline runtime/src/bucket_map_holder.rs:12]: Is there already a slot time constant somewhere? Or, is the intent to have the Age time be tunable and different from the slot time?

File: runtime/src/bucket_map_holder.rs [hunk 2]
```diff
@@ -53,6 +59,12 @@ impl<T: IndexValue> BucketMapHolder<T> {
     pub fn future_age_to_flush(&self) -> Age {
         self.current_age().wrapping_add(self.ages_to_stay_in_cache)
     }
+
+    fn has_age_interval_elapsed(&self) -> bool {
+        // note that when this returns true, state of age_timer is modified
+        self.age_timer.should_update(AGE_MS) // todo: make sure this interval has correct Ordering
```
Comment (brooksprumo) [inline runtime/src/bucket_map_holder.rs:65]: Should this todo be resolved before the PR gets merged?

Comment (jeffwashington) [reply runtime/src/bucket_map_holder.rs:65]: I was hoping you'd help me resolve this. I added tests and it appears to work. should_update uses Ordering::Relaxed.

File: runtime/src/bucket_map_holder.rs (hunk 1 above)
Comment (jeffwashington) [reply runtime/src/bucket_map_holder.rs:12]: I wanted it different, but logically the same. Maybe it does make sense to use the clock constant. I'll investigate.

Comment (jeffwashington) [reply runtime/src/bucket_map_holder.rs]: I made it the same constant

File: runtime/src/bucket_map_holder.rs (hunk 2 above)
Comment (jeffwashington) [reply runtime/src/bucket_map_holder.rs]: I got rid of the comment for now. Please do look at AtomicInterval and see if the orderings make sense.

Comment (brooksprumo) [reply runtime/src/bucket_map_holder.rs]: Gotcha, I'll take a look

Comment (brooksprumo) [reply runtime/src/bucket_map_holder.rs]: I think this looks good

**Review by brooksprumo — APPROVED** [review]

**Pull request merged**